sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -785,13 +785,36 @@ class CodegenContext {
    * @param expressions the codes to evaluate expressions.
    */
   def splitExpressions(row: String, expressions: Seq[String]): String = {
-    if (row == null || currentVars != null) {
+    if (INPUT_ROW == null || currentVars != null) {
       // Cannot split these expressions because they are not created from a row object.
       return expressions.mkString("\n")
     }
     splitExpressions(expressions, funcName = "apply", arguments = ("InternalRow", row) :: Nil)
   }
 
+  /**
+   * Splits the generated code of expressions into multiple functions, because function has
+   * 64kb code size limit in JVM. This version takes care of INPUT_ROW and currentVars
+   *
+   * @param expressions the codes to evaluate expressions.
+   * @param funcName the split function name base.
+   * @param argumentsExceptRow the list of (type, name) of the arguments of the split function
+   *                           except for ctx.INPUT_ROW
+  */
+  def splitExpressions(
+      expressions: Seq[String],
+      funcName: String,
+      argumentsExceptRow: Seq[(String, String)]): String = {
```

Comment:
Let the caller also provides `ctx.INPUT_ROW`?
Change it to
```Scala
arguments: Seq[(String, String)]
```

Reply:
I agree that it is good from the view of consistency. I have one question in my mind.

If we use the same argument name `arguments`, is it possible to for developer to distinguish this `splitExpressions` from the below (rich) `splitExpressions` when they want to pass only three arguments `expressions`, `funcName`, and `arguments`?

Reply:
Could we combine them in the same function?

Reply:
Now, we have three `splitExpressions` in this PR.
1. `splitExpression(row, expressions)`
2. `splitExpression(expressions, funcName, arguments)`
3. `splitExpression(expressions, funcName, arguments, returnType, ...)`

It is hard to combine 2. and 3. since 2. takes care of `INPUT_ROW` and `currentVars` while 3. does not take care of them.
Are you suggesting me to combine 1. and 2. which take care of `INPUT_ROW` and `currentVars`?

Reply:
Could you check the caller of case 3 which also need to check `INPUT_ROW` and `currentVars`? It sounds like some of them miss the checking. 

In addition, case `1` and `2` can be easily combined. 

I think we need a different name for case `1` and `2`. How about `splitExpressionsOnInputRow`?

cc @cloud-fan

Reply:
I see. I will check it tonight.

Reply:
I confirmed there are some cases that do not require to check `INPUT_ROW` and `currentVars`.
- access fields in struct
- `UnsafeJoiner`
- comparison for ordering

I will try to merge cases `1` and `2`. If a different name is required, I will use `splitExpressionsOnInputRow`.